framework-plugins/lisk-framework-chain-connector-plugin/src/db.ts

```diff
@@ -36,59 +43,124 @@ export const getDBInstance = async (
 	return new Database(dirPath);
 };
 
-export const getChainConnectorInfo = async (db: KVStore): Promise<ChainConnectorInfo> => {
-	try {
-		const encodedInfo = await db.get(EMPTY_BYTES);
-		return codec.decode<ChainConnectorInfo>(chainConnectorInfoSchema, encodedInfo);
-	} catch (error) {
-		debug('Chain connector info does not exist.');
-		return {
-			blockHeaders: [],
-			aggregateCommits: [],
-			validatorsHashPreimage: [],
-			crossChainMessages: [],
-		};
+export class ChainConnectorStore {
+	private readonly _db: KVStore;
+	public constructor(db: KVStore, private readonly _chainType: Buffer) {
+		this._db = db;
 	}
-};
 
-export const setChainConnectorInfo = async (
-	db: KVStore,
-	chainConnectorInfo: ChainConnectorInfo,
-): Promise<void> => {
-	const encodedInfo = codec.encode(chainConnectorInfoSchema, chainConnectorInfo);
-	await db.set(EMPTY_BYTES, encodedInfo);
-};
+	public close() {
```

Comment:
This method can be used to do some extra steps for cleanup if needed, and since DAO is using the DB directly then it can be used to close it, the only part which is inconsistent is that we initialize DB is the plugin class so it should also be closed in that class. Lets keep it as it is right now and we can come back to it once we have a more stable code of the plugin